bot/admin/admin.py

```diff
@@ -348,6 +353,102 @@ async def change_discord_presence_clear(self, _ctx: commands.Context):
         """
         await self.bot.change_presence(activity=None)
 
+    @commands.command(name="botonly")
+    @command_log
+    async def botonly(self, ctx: commands.Context, channel: Optional[discord.TextChannel]):
+        """Command handler for the `botonly` command.
+
+        This command marks a channel in the database as bot-only, so every message posted by someone else than the bot
+        will be deleted immediately.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            channel (discord.Textchannel): The channel that is to be made bot-only
+        """
+        target_channel = channel if channel is not None else ctx.channel
+        is_channel_botonly = self._db_connector.is_botonly(target_channel)
+        if is_channel_botonly:
+            log.info("Deactivated bot-only mode for channel {0}".format(target_channel))
+            self._db_connector.deactivate_botonly(target_channel)
+        else:
+            log.info("Activated bot-only mode for channel {0}".format(target_channel))
+            self._db_connector.activate_botonly(target_channel)
+
+        is_enabled_string = 'aktiviert' if not is_channel_botonly else 'deaktiviert'
+        embed = _build_botonly_embed(is_enabled_string)
+        await target_channel.send(embed=embed)
+
+    # todo comment in when issue demands so
+
+    # @commands.command(name="purge")
+    # @command_log
+    # async def purge_channel(self, ctx: commands.Context, channel: discord.TextChannel):
+    #     """Command handler for the `purge` command.
+    #
+    #     Removes all messages in a channel.
+    #
+    #     Args:
+    #         ctx (discord.ext.commands.Context): The context from which this command is invoked.
+    #         channel (discord.Textchannel): The channel that is to purge
+    #     """
+    #     embed = _build_purge_confirmation_embed(channel)
+    #     timeout = 15.0
+    #     reaction = await self._send_confirmation_dialog(ctx, embed, timeout)
+    #     if reaction is None:
+    #         return
+    #     if str(reaction[0].emoji) == constants.EMOJI_CONFIRM:
+    #         await _purge_channel(channel)
+
+    @commands.Cog.listener(name='on_message')
+    async def on_message(self, ctx: discord.Message):
+        """Event Handler for new messages.
+
+        Deletes a message if the channel it was posted in is in bot-only mode and the author isn't SAM.
+
+        Args:
+            ctx (discord.Message): The context this method was called in. Must always be a message.
+        """
+        if ctx.author == self.bot.user:
```

Comment:
How about:
```python
if self._db_connector.is_botonly(ctx.channel) and ctx.author != self.bot.user:
    await ctx.delete()
```

Reply:
bc if we made more checks for this event listener I want to globally check that it is not a bot message first.

Eg. if we counted the number of spam. messages, or checked for bad words I want to exclude bot messages always. This way I don't have to think about it in this cog anymore if I extend it